### PR TITLE
Docathon! Inst Sched Map (and CmdDef)

### PR DESCRIPTION
**Pull request by lcapelluto**
### Summary

Docathon: Update the documentation TOC, cmd_def module docstring, and InstructionScheduleMap docstrings

### Details and comments

I verified changes on a local docs build

### CODE REVIEW

File: qiskit/pulse/instruction_schedule_map.py [hunk 1]
```diff
@@ -80,25 +88,24 @@ def qubits_with_instruction(self, instruction: str) -> List[Union[int, Tuple[int
...
             qubits: A qubit index, or a list or tuple of indices.
 
         Returns:
-            All the instructions which are defined on the qubits. For 1 qubit, all the 1Q
-            instructions defined. For multiple qubits, all the instructions which apply to that
-            whole set of qubits (e.g. qubits=[0, 1] may return ['cx']).
+            All the instructions which are defined on the qubits.
+
+            For 1 qubit, all the 1Q instructions defined. For multiple qubits, all the instructions
+            which apply to that whole set of qubits (e.g. qubits=[0, 1] may return ['cx']).
```
Comment (DanPuzzuoli) [inline qiskit/pulse/instruction_schedule_map.py]: Should qubits=[0,1] and ['cx'] be in the code quotes?

**Comment by DanPuzzuoli** [comment]
Aside from the one comment looks good to me!

**Comment by lcapelluto** [comment]
@DanPuzzuoli  thanks for review! Updated! 😄

File: qiskit/pulse/instruction_schedule_map.py [hunk 2]
```diff
@@ -174,17 +175,13 @@ def add(self,
             instruction: str,
             qubits: Union[int, Iterable[int]],
             schedule: [Schedule, ParameterizedSchedule]) -> None:
```
Comment (DanPuzzuoli) [inline qiskit/pulse/instruction_schedule_map.py]: I think this should be `Union[Schedule, ParameterizedSchedule]`

File: qiskit/pulse/instruction_schedule_map.py [hunk 3]
```diff
@@ -38,8 +38,9 @@
 
 
 class InstructionScheduleMap():
-    """Mapping from QuantumCircuit Instruction names and qubits to Schedules. In particular, the
-    mapping is formatted as type::
+    """Mapping from :py:class:`qiskit.circuit.QuantumCircuit`:py:class:`qiskit.circuit.Instruction`
```
[the remark this answers is not in the document]
Comment (lcapelluto) [reply qiskit/pulse/instruction_schedule_map.py]: yay this is great! I did not know!

**Pull request merged**